Pancakeswap-Agents/Large-LP-Token-Deposit-Withdraw/src/masterchef.fetcher.ts

```diff
@@ -0,0 +1,19 @@
+import { providers, Contract, BigNumber } from "ethers";
+import { MASTERCHEF_ABI } from "./constants";
+
+export default class MasterchefFetcher {
+
+    readonly provider: providers.Provider;
+    private masterchefContract: Contract;
+
+    constructor(provider: providers.Provider, masterchefAddress: string) {
+        this.provider = provider;
+        this.masterchefContract = new Contract(masterchefAddress, MASTERCHEF_ABI, provider);
+    }
+
+    public async getLPToken(pid: string | number, block: string | number) {

```

Comment:
```suggestion
    public async getLPToken(pid: BigNumber, block: string | number) {
```
That's the correct type, right?